feat: ignore nore on <Plug> maps (#16969)
neovim/neovim: runtime/doc/vim_diff.txt

- `getcwd(-1)` is equivalent to `getcwd(-1, 0)` instead of returning the global
  working directory. Use `getcwd(-1, -1)` to get the global working directory.

> Keymap:
> - |nore| isn't applicable to <Plug> keymaps. <Plug> keymaps are always remaped.

Reviewer: Personally I prefer "mappings" to "keymaps", especially in `vim_diff.txt`, as in Vim's terminology "mapping" and "keymap" are two things.
```suggestion
Mappings:
- |nore| is ignored for rhs <Plug> mappings. <Plug> mappings are always remapped.
```
BTW, paragraphs in this section are sorted alphabetically by their first line.

Author: > I feel this should also be documented in vim_diff.txt.

Sorry ? Paragraphs are sorted based on first line ? I don't I understood that part.

Reviewer: That part is not important. You can ignore that.

Author: Done.
